src/py_pecg/filtering_module.py

```diff
@@ -0,0 +1,564 @@
+
+    if fs == 500:
+        # Interpolate by 2
+        interp_params = [(2, 1, 0), (2, 3, 0), (2, 7, 0), (2, 7, 0), (2, 7, 0)]
+        quadratic_filter_array = apply_interpolation(
+            quadratic_filter_array, interp_params
+        )
+        for i in range(0, len(quadratic_filter_array)):
+            quadratic_filter_array[i] = quadratic_filter_array[i][1:-2]
+
+    elif fs == 1000:
+        # Interpolate by 4
+        interp_params = [(4, 1, 0), (4, 3, 0), (4, 7, 0), (4, 7, 0), (4, 7, 0)]
+        quadratic_filter_array = apply_interpolation(
+            quadratic_filter_array, interp_params
+        )
+        for i in range(0, len(quadratic_filter_array)):
+            quadratic_filter_array[i] = quadratic_filter_array[i][1:-4]
+
+    elif fs == 200:
+        # Interpolate by 4
+        interp_params = [
+            False,
+            (4, 3, 0.4),
+            (4, 7, 0.4),
+            (4, 7, 0.4),
+            (4, 7, 0.4),
+        ]
+        quadratic_filter_array = apply_interpolation(
+            quadratic_filter_array, interp_params
+        )
+        for i in range(0, len(quadratic_filter_array)):
+            quadratic_filter_array[i] = quadratic_filter_array[i][1:-4]
+
+        # Downsample by 5
+        quadratic_filter_array[0] = 1.1 * np.array([5 / 4, -5 / 4])
+        for i, ind in enumerate([False, 3, 4, 1, 1]):
+            if i != 0:
+                quadratic_filter_array[i] = quadratic_filter_array[i][ind::5]
+
+    elif fs == 360:
+        # Interpolate by 6
+        interp_params = [(6, 1, 0), (6, 3, 0), (6, 7, 0), (6, 7, 0), (6, 7, 0)]
+        quadratic_filter_array = apply_interpolation(
+            quadratic_filter_array, interp_params
+        )
+        for i in range(0, len(quadratic_filter_array)):
+            quadratic_filter_array[i] = quadratic_filter_array[i][1:-6]
+
+        # Downsample by 5
+        for i, ind in enumerate([3, 0, 4, 2, 2]):
+            quadratic_filter_array[i] = quadratic_filter_array[i][ind::5]
+
+        # Interpolate by 6 again
+        interp_params = [(6, 1, 0), (6, 3, 0), (6, 7, 0), (6, 7, 0), (6, 7, 0)]
+        quadratic_filter_array = apply_interpolation(
+            quadratic_filter_array, interp_params
+        )
+        for i in range(0, len(quadratic_filter_array)):
+            quadratic_filter_array[i] = quadratic_filter_array[i][1:-6]
+
+        # Final Downsample by 5
+        for i, ind in enumerate([1, 4, 3, 3, 3]):
+            quadratic_filter_array[i] = quadratic_filter_array[i][ind::5]
+
+    return quadratic_filter_array, messages
+
+
+def wavelet_transform(
+    signal: np.ndarray,
+    quadratic_filter_array: np.ndarray,
+) -> np.ndarray:
+    """
+    Calculates the wavelet transform of a signal using quadratic spline wavelet.
+    It calculates wavelets in scales of 1 to 4.
+
+    Args:
+        signal (np.ndarray): The target signal designated for filtering.
+        quadratic_filter_array (np.ndarray): Qquadratic spline filter bank.
```

Comment:
You assume the length of the `quadratic_filter_array` to be 5 in the code, but don't say it in the docstring. BTW, the `quadratic_filter_array ` created by `quadratic_splines_filterbank` is a list of arrays not an array